src/signalr-hub.spec.ts

```diff
@@ -77,12 +77,68 @@ describe('SignalRHub', () => {
         expect(console.warn).not.toHaveBeenCalled();
     });
 
-    it('should send data to proxy', () => {
+    it('should send data to proxy', async () => {
+        spyOn(proxy, 'invoke').and.returnValue(Promise.resolve());;
```

Comment:
This is repeated in most tests. Move to beforeEach?

Reply:
I don't think that this is a good idea. There are five spies on the `invoke`-method of the proxy. Three of them configure the spy to return a resolved promise, one configures the spy to return nothing at all and one configures the spy to return a resolved proxy with a value.

All other tests don't need that proxy at all.

As you cannot configure a proxy twice as far as I know this would make the test implementation more complex than needed I think.